owrrm/src/main/java/com/facebook/openwifi/rrm/optimizers/clientsteering/ClientSteeringState.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.openwifi.rrm.optimizers.clientsteering;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/** Singleton class to manage global client steering state */
+public final class ClientSteeringState {
+
+	/** Singleton instance */
+	private static final ClientSteeringState singleton =
+		new ClientSteeringState();
+
+	/** Private default constructor (this is a singleton class) */
+	private ClientSteeringState() {};
+
+	/** Get the singleton instance */
+	public static ClientSteeringState getInstance() {
+		return singleton;
+	}
+
+	/**
+	 * Map from AP serial number to client MAC to time (JVM monotonic time in
+	 * ns) of the latest attempted client steering action. The {@code Long}
+	 * values are never null.
+	 */
+	private ConcurrentMap<String, Map<String, Long>> apClientLastAttempt =
+		new ConcurrentHashMap<>();
+
+	/** Reset the state - ONLY FOR TESTING */
+	public synchronized void reset() {
+		// TODO better way then to call this before every test??
+		apClientLastAttempt.clear();
+	}
+
+	/**
+	 * Register a client steering attempt for the given AP and station at the
```

Comment:
This is a little weird to me because you are registering the attempt in a different section of code from actually executing the attempt. That is fine, but please make it more clear that this is only checking the **registration time** not the actual execution time.

Also, looks like this doesn't work in the "dry run" situation. Please pass that flag down.